### PR TITLE
Test cases for passport bearer strategy

### PR DESCRIPTION
**Pull request by SFIN571**

### CODE REVIEW

File: src/__tests__/fixtures/sequences/authentication.sequence.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      //call authentication action
+      await this.authenticateRequest(request);
+
+      // Authentication successful, proceed to invoke controller
+      const args = await this.parseParams(request, route);
+      const result = await this.invoke(route, args);
+      this.send(response, result);
+    } catch (error) {
+      //
+      // The authentication action utilizes a strategy resolver to find
```
Comment (samarpan-b) [inline src/__tests__/fixtures/sequences/authentication.sequence.ts]: remove this comment or make it more meaningful. We cannot have @loopback comments here

File: src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import {authenticate} from '../../../decorators';
+import {STRATEGY} from '../../../strategy-name.enum';
+import {getApp} from './helpers';
+import {MyAuthenticationSequence} from '../../fixtures/sequences/authentication.sequence';
+import {Strategies} from '../../../strategies/keys';
+
+/**
+ * Testing overall flow of authentication with bearer strategy
+ */
+describe('getting strategy with options', () => {
```
Comment (samarpan-b) [inline src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts]: can we check and follow the description pattern from loopback repo ?

File: src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  let app: Application;
+  let server: RestServer;
+  beforeEach(givenAServer);
+  beforeEach(givenAuthenticatedSequence);
+  beforeEach(getAuthVerifier);
+
+  it('should return 401 bad request', async () => {
```
Comment (samarpan-b) [inline src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts]: when should this happen ?

File: src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    app.controller(TestController);
+
+    await whenIMakeRequestTo(server)
+      .get('/test')
+      .expect(401);
+  });
+
+  it('should return status 200', async () => {
```
Comment (samarpan-b) [inline src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts]: when ?

File: src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    app.controller(TestController);
+
+    await whenIMakeRequestTo(server)
+      .get('/test')
+      .set('Authorization', 'sometoken')
+      .expect(401);
+  });
+
+  function whenIMakeRequestTo(restServer: RestServer): Client {
```
Comment (samarpan-b) [inline src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts]: define all the helper classes and function in a separate file

File: src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      }
+    }
+
+    app.controller(TestController);
+
+    await whenIMakeRequestTo(server)
+      .get('/test')
+      .set('Authorization', 'sometoken')
+      .expect(401);
+  });
```
Comment (samarpan-b) [inline src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts]: where is the TC for checking if responded with a user model ?

File: src/__tests__/unit/passport-bearer/bearer-token-verify.unit.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  BearerStrategyFactoryProvider,
+  BearerStrategyFactory,
+} from '../../../strategies';
+import * as PassportBearer from 'passport-http-bearer';
+import {expect} from '@loopback/testlab';
+
+/**
+ * Testing to get bearer strategy from providers
+ */
+describe('getting strategy with options', () => {
```
Comment (samarpan-b) [inline src/__tests__/unit/passport-bearer/bearer-token-verify.unit.ts]: check text

File: src/__tests__/unit/passport-bearer/bearer-token-verify.unit.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+    const options: PassportBearer.IStrategyOptions = {
+      scope: '',
+      realm: '',
+      passReqToCallback: false,
+    };
+
+    const bearerStrategyVerifier = strategyVerifier(options);
+
+    expect(bearerStrategyVerifier).to.have.property('name');
```
Comment (samarpan-b) [inline src/__tests__/unit/passport-bearer/bearer-token-verify.unit.ts]: what is the difference between these 2 TCs ?

**Comment by samarpan-b** [comment]
Please add description to your PR please

File: src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts (hunk 2 above)
Comment (SFIN571) [reply src/__tests__/integration/passport-bearer/bearer-token-verify.integration.ts]: they have not commented on any of their test cases

**Pull request merged**